docs/api-docs/apps/tutorials/sample-app-next/06-step-4-enhance-ux.md

```diff
@@ -0,0 +1,797 @@
+
+</div>
+</div>
+</div>
+
+3. Add the `Products` functional component.
+
+```js
+const Products = () => {
+    const router = useRouter();
+    const { isError, isLoading, list = [] } = useProductList();
+    const tableItems = list.map(({ id, inventory_level: stock, name, price }) => ({
+        id,
+        name,
+        price,
+        stock,
+    }));
+
+    const renderName = (id: number, name: string): ReactElement => (
+        <Link href={`/products/${id}`}>
+            <StyledLink>{name}</StyledLink>
+        </Link>
+    );
+
+    const renderPrice = (price: number): string => (
+        new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(price)
+    );
+
+    const renderStock = (stock: number): ReactElement => (stock > 0
+        ? <Small>{stock}</Small>
+        : <Small bold marginBottom="none" color="danger">0</Small>
+    );
+
+    const renderAction = (id: number): ReactElement => (
+        <Dropdown
+            items={[ { content: 'Edit product', onItemClick: () => router.push(`/products/${id}`), hash: 'edit' } ]}
+            toggle={<Button iconOnly={<MoreHorizIcon color="secondary60" />} variant="subtle" />}
+        />
+    );
+
+    return (
+        <Panel>
+            <StatefulTable
+                columns={[
+                    { header: 'Product name', hash: 'name', render: ({ id, name }) => renderName(id, name), sortKey: 'name' },
+                    { header: 'Stock', hash: 'stock', render: ({ stock }) => renderStock(stock), sortKey: 'stock' },
+                    { header: 'Price', hash: 'price', render: ({ price }) => renderPrice(price), sortKey: 'price' },
+                    { header: 'Action', hideHeader: true, hash: 'id', render: ({ id }) => renderAction(id), sortKey: 'id' },
+                ]}
+                items={tableItems}
+                itemName="Products"
+                stickyHeader
+            />
+        </Panel>
+    );
+};
+
+export default Products;
+```
+
+The `Products` functional component uses the BigDesign's [StatefulTable](https://developer.bigcommerce.com/big-design/statefulTable), a wrapper of the [Table](https://developer.bigcommerce.com/big-design/table) component that supports pagination, row selection, and sorting.
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-4-big-design/pages/products/index.tsx)
+
+## Add the InnerHeader component
+
+1. In the `components` folder, create an `innerHeader.tsx` file.
+
+2. At the top of the file, import the following packages:
+
+```js
+import { Box, Button, H1, HR, Text } from '@bigcommerce/big-design';
+import { ArrowBackIcon } from '@bigcommerce/big-design-icons';
+import { useRouter } from 'next/router';
+import { useProductList } from '../lib/hooks';
+import { TabIds, TabRoutes } from './header';
+```
+
+3. Define the `InnerHeader` functional component. 
```

Comment:
we might want to explain that the `InnerHeader` component is used for the product edit page, which is a subpage of `products` (/products/[pid]), whereas the main `Header` component is used for the main pages, '/' and '/products'